sys/shell/commands/sc_gnrc_icmpv6_echo.c

```diff
@@ -171,21 +171,23 @@ static int _configure(int argc, char **argv, _ping_data_t *data)
     for (int i = 1; i < argc; i++) {
         char *arg = argv[i];
         if (arg[0] != '-') {
+            int iface;
+
             data->hostname = arg;
 #ifdef MODULE_SOCK_DNS
             if (sock_dns_query(data->hostname, &data->host, AF_INET6) == 0) {
                 continue;
             }
 #endif
-            data->iface = ipv6_addr_split_iface(data->hostname);
-            if (data->iface < KERNEL_PID_UNDEF) {
+            iface = ipv6_addr_split_iface(data->hostname);
+            if (iface > KERNEL_PID_UNDEF) {
```

Comment:
The doc of [`ipv6_addr_split_iface()`](http://api.riot-os.org/group__net__ipv6__addr.html#ga781811ddaaa0f3a327cfaa306ea26769) indicates that `-1` would be the return value that indicates an error, not `KERNEL_PID_UNDEF` (which is defined to be `0`). I'd assume that `ipv6_addr_split_iface()` will never return 0, so this would behave identical with checking for `-1`. But I'd prefer if the code would check for the error code given in the API documentation.

Reply:
As you can see, this was just adapted from the existing code. However, looking at other calls of `ipv6_addr_split_iface()` backs you. So I'll provide a piggy-back fix for that.

Reply:
See https://github.com/RIOT-OS/RIOT/pull/11916/commits/d1abb29665d7f73eb7ae6ca3a45f107b528fe88e. I would recommend to redo the test though (don't have the hardware at hand to to so myself atm)

Reply:
> As you can see, this was just adapted from the existing code.

I'm aware :-D

I changed the indent level of existing code in a PR recently and had to fix all the issues of the existing code then :-D